test/common/http/http2/metadata_encoder_decoder_test.cc

```diff
@@ -0,0 +1,356 @@
+#include "common/buffer/buffer_impl.h"
+#include "common/common/logger.h"
+#include "common/http/http2/metadata_decoder.h"
+#include "common/http/http2/metadata_encoder.h"
+#include "common/runtime/runtime_impl.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "nghttp2/nghttp2.h"
+
+// A global variable in nghttp2 to disable preface and initial settings for
+// tests.
+extern int nghttp2_enable_strict_preface;
```

Comment:
Is this the only way to do this? There is no option that can be set directly on the client? If not can we get an upstream issue tracking adding such an option?

Reply:
Great suggestion! Turns out there is an option! The global variable is removed.

Reply:
Sorry did you mean for this to be back now? I think maybe some code got reverted?

Reply:
Sorry about the iteration. The option only allows nghttp2 to skip checking connection preface but still checking initial settings. We would like to skips both in test, so I added the global variable back. I can see the option will be handy in test. I will open an bug with nghttp2.

Reply:
OK. Can you add a TODO comment here w/ a link to the upstream nghttp2 issue?